clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -150,6 +150,10 @@ private boolean usingMemoryPool() {
     }
 
     public void incRefCount() {
+        if (!usingMemoryPool()) {
+            return;
+        }
+
         if (bufferReleased && usingMemoryPool()) {
```

Comment:
is usingMemoryPool needed here? should always be true? same in decRefCnt

Reply:
Good catch!